kratos/solving_strategies/strategies/explicit_solving_strategy_runge_kutta.h

```diff
@@ -34,6 +36,224 @@ namespace Kratos
 ///@name Type Definitions
 ///@{
 
+/** @brief Butcher tableau for Runge-Kutta method.
+ *
+ * Contains all info necessary of a particular RK method.
+ * It specifies the coefficients of the particular Runge-Kutta method.
+ *
+ * @tparam: Derived:       A derived class that must contain the methods specified below.
+ * @tparam: TOrder:        The order of integration.
+ * @tparam: TSubstepCount: The number of sub-steps.
+ *
+ * Child class must provide methods:
+ * - static const MatrixType GenerateRKMatrix() -> Provides coefficients a_ij
+ * - static const VectorType GenerateWeights()  -> Provides coefficients b_i
+ * - static const VectorType GenerateNodes()    -> Provides coefficients c_i
+ * - static std::string Name()
+ */
+template<typename Derived, unsigned int TOrder, unsigned int TSubstepCount>
+class ButcherTableau
+{
+public:
+    static constexpr unsigned int Order() {return TOrder;}
+    static constexpr unsigned int SubstepCount() {return TSubstepCount; }
+
+    typedef array_1d<double, SubstepCount()> VectorType;
+    typedef std::array<array_1d<double, SubstepCount()-1>, SubstepCount()-1> MatrixType;
```

Comment:
Wouldn't it be easier to use a `BoundedMatrix<double,TSubstepCount-1,TSubstepCount-1>`?

Reply:
A is never used in its matrix form, only row-by-row. Hence constructing rows every time we need them (once per dof, per substep, per time-step) is unnecessary and we could simply have the rows form the get-go.

Reply:
Scratch that, with recent changes this is much less of an issue. Moving back to BoundedMatrix.